.github/workflows/ci-additional.yaml

```diff
@@ -157,6 +157,51 @@ jobs:
         run: |
           python -m pytest --doctest-modules xarray --ignore xarray/tests
 
+  typing:
+    name: Type checking (mypy)
+    runs-on: "ubuntu-latest"
+    needs: detect-ci-trigger
+    if: needs.detect-ci-trigger.outputs.triggered == 'false'
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # Fetch all history for all branches and tags.
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          channels: conda-forge
+          channel-priority: strict
+          mamba-version: "*"
+          activate-environment: xarray-tests
+          auto-update-conda: false
+          python-version: "3.8"
+
+      - name: Install conda dependencies
+        run: |
+          mamba env update -f ci/requirements/environment.yml
+      - name: Install mypy
+        run: |
+          # version must correspond to the one in .pre-commit-config.yaml
+          mamba install mypy=0.800
```

Comment:
@keewis, is there any way we can automatically keep this version in sync with changes that will be introduced by https://github.com/pydata/xarray/blob/master/.github/workflows/ci-pre-commit-autoupdate.yaml? In other words, can we make it easy to update this version whenever `.pre-commit-config.yaml` is updated?

Reply:
sure. The action used to create the PR allows specifying any number of commands, so we can add a script which does the syncing. That might be easier if we create a new environment file (`requirements.txt` like?) containing just `mypy`, though.